packages/opentelemetry-api/src/diag/logLevel.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DiagAPI } from '../api/diag';
+import { Logger } from '../common/Logger';
+import { DiagLogger, DiagLogFunction, createNoopDiagLogger } from './logger';
+
+/**
+ * Defines the available internal logging levels for the diagnostic logger, the numeric values
+ * of the levels are defined to match the original values from the initial LogLevel to avoid
+ * compatibility/migration issues for any implementation that assume the numeric ordering.
+ */
+export enum DiagLogLevel {
```

Comment:
why do we have so many different level ? Can't we use error/warn/info/debug like most logger do ? As a end user i directly give the library that i use to log (pino in my case) to otel, having some many levels (that i dont personally think we really need) means that i will need to do some code to translate those into level my logger understand.

Reply:
The comment on the number of logging levels depends on your history, most of the logging systems I have used (and created) actually have more than this. .Net, log4j. 
As a diagnostic logger, especially for a system where we don't want to publish new version regularly (esp once we become stable) it is extremely useful to have multiple levels to diagnose internal failures for different scenarios -- without needing to debug with the end-user or release a "special" version with extra debugging (diagnostic logging) included.

Reply:
> The comment on the number of logging levels depends on your history, most of the logging systems I have used (and created) actually have more than this. .Net, log4j.

I understand but don't think its a valid point, the goal is not have a many level as possible.

> As a diagnostic logger, especially for a system where we don't want to publish new version regularly (esp once we become stable) it is extremely useful to have multiple levels to diagnose internal failures for different scenarios -- without needing to debug with the end-user or release a "special" version with extra debugging (diagnostic logging) included.

Again i understand that we need to avoid publishing the api often but i stand my point: its about having just what we need. 

For exemple, why differenciate terminal/critial and error which is generic enough, the level doesn't provide much (specially since it will be mostly likely serialized as a number) what is important is the error message.
Also what the difference between verbose and debug ? currently debug is used for what you describe as verbose

I don't see the point of having a lot of level "in case we might need them", specially since my original problem is still there, the less level we have the more likely it can be plugged with whatever logger the end user is already using

Reply:
I had the same thought about the number of log levels but came to the same conclusion. It may be difficult to add levels later if we need them and this gives us a fair bit of control when trying to debug. One change I might make is to make the levels `10, 20, 30, ...` instead of `1, 2, 3, ...`. This would make it much easier to insert a log level between existing levels (like startupinfo) later as level 25 or 35 or something without breaking deployed loggers.

Reply:
> why differenciate terminal/critial and error

I think I added this as the description on the levels, terminal and critical are important level events

> Also what the difference between verbose and debug 

Again see the descriptions, for history ```verbose``` was originally called ``` trace```

> I don't see the point of having a lot of level "in case we might need them"

They are not "in case", when used properly they are extremely useful.

> the less level we have the more likely it can be plugged with whatever logger the end user is already using

This was considered and the LogLevel logger handles this [here](https://github.com/open-telemetry/opentelemetry-js/pull/1880/files#diff-e6f7471e2882259590c8af49464f67eea88c18377866cdc0a6952ca1bbb38239R74-R83) as part of the followup PR to remove the Logger (which will be a breaking change) this can either be removed or retained to support this scenario.

> One change I might make is to make the levels ```10, 20, 30, ...```

I did originally consider this (as part of the pre-decessor PR to this one), however, to keep backward compatibility the numeric value of the existing LogLevel and DiagLogLevel where matched -- As this would be a breaking change, I'm happy to make this change as part of the followup breaking PR.

Timing wise, my plan was to publish the followup breaking change PR after we publish the next API release, so that this can be "waiting" for the next release and in the meantime I can also take the task to update Contrib.

Reply:
Which timeline wise (I'm assuming) would be after @dyladan has created the new Api Repo.

Reply:
Adding ranges (0..10..20..) as there was no "space" for the Startup level, this will break numeric compatibility between LogLevel and DiagLogLevel -- this WILL cuase some confusion and compatibility issues with the Environment OTEL_LOG_LEVEL -- But this will be "fixed" as part of the followup PR.

Reply:
> But, as stated below, it is easier to include levels now we may not need later than it is to add them later.

(answering here for @dyladan comment [here](https://github.com/open-telemetry/opentelemetry-js/pull/1880#discussion_r573203816)) : I think this drill down to a different opinion about software design, i would prefer have basic levels (like said [here](https://github.com/open-telemetry/opentelemetry-js/pull/1880#issuecomment-771128083)) and design so we can add more later (0..10..20 are a good way to do this) only if we fill the need for more.

Anyway if the consensus is more to have a lot in the first place i don't mind.